lib/findtoken.h

```diff
@@ -74,105 +74,33 @@ T* findToken(T* start, const Token* end, const Predicate& pred)
-                }
-            } else {
-                if (findTokensSkipDeadCodeImpl(library, thenStart, thenStart->link(), pred, found, evaluate, skipUnevaluated))
-                    return true;
-                if (isReturnScope(thenStart->link(), library))
-                    return true;
-                tok = thenStart->link();
-            }
-        } else if (Token::Match(tok->astParent(), "&&|?|%oror%") && astIsLHS(tok)) {
-            auto result = evaluate(tok);
-            if (result.empty())
-                continue;
-            const bool cond = result.front() != 0;
-            T* next = nullptr;
-            if ((cond && Token::simpleMatch(tok->astParent(), "||")) ||
-                (!cond && Token::simpleMatch(tok->astParent(), "&&"))) {
-                next = nextAfterAstRightmostLeaf(tok->astParent());
-            } else if (Token::simpleMatch(tok->astParent(), "?")) {
-                T* colon = tok->astParent()->astOperand2();
-                if (!cond) {
-                    next = colon;
-                } else {
-                    if (findTokensSkipDeadCodeImpl(library, tok->astParent()->next(), colon, pred, found, evaluate, skipUnevaluated))
-                        return true;
-                    next = nextAfterAstRightmostLeaf(colon);
-                }
-            }
-            if (next)
-                tok = next;
-        } else if (Token::simpleMatch(tok, "} else {")) {
-            const Token* condTok = getCondTokFromEnd(tok);
-            if (!condTok)
-                continue;
-            auto result = evaluate(condTok);
-            if (result.empty())
-                continue;
-            if (isReturnScope(tok->link(), library))
-                return true;
-            auto r = result.front();
-            if (r != 0) {
-                tok = tok->linkAt(2);
-            }
-        } else if (Token::simpleMatch(tok, "[") && Token::Match(tok->link(), "] (|{")) {
-            T* afterCapture = tok->link()->next();
-            if (Token::simpleMatch(afterCapture, "(") && afterCapture->link())
-                tok = afterCapture->link()->next();
-            else
-                tok = afterCapture;
-        }
-        if (skipUnevaluated && isUnevaluated(tok)) {
-            T *next = tok->linkAt(1);
-            if (!next)
-                continue;
-            tok = next;
-        }
-    }
-    return false;
+namespace internal {
+    template<class T>
+    bool findTokensSkipDeadCodeImpl(const Library &library,
+                                    T *start,
+                                    const Token *end,
+                                    const std::function<bool(const Token *)> &pred,
+                                    const std::function<bool(T *)>& found,
+                                    const std::function<std::vector<MathLib::bigint>(const Token *)> &evaluate,
+                                    bool skipUnevaluated) = delete;
```

Comment:
Why does this need to be deleted?

Reply:
Because we are only using specializations. If this is not deleted the compiler will complain about it missing.

Reply:
Ah, I see. Using templates makes no sense anymore everything related to it can just be dropped.

Reply:
Done.